distribution/lib/Standard/Database/0.0.0-dev/src/Internal/IR.enso

```diff
@@ -1,326 +1,10 @@
-         parts of the query, this is especially useful for example in
-         self-joins, allowing to differentiate between different instances of
-         the same table.
-    From_Query (raw_sql : Text) (alias : Text)
-
-    ## PRIVATE
-
-       A query source that performs a join operation on two sources.
-
-       Arguments:
-       - kind: the kind of the join.
-       - left_spec: the left data source to be joined
-       - right_spec: the right data source to be joined
-       - on: a list of expressions that will be used as join conditions, these
-         are usually be equalities between expressions from the left and right
-         sources.
-    Join (kind : Join_Kind) (left_spec : From_Spec) (right_spec : From_Spec) (on : Vector Expression)
-
-    ## PRIVATE
-
-       A query source consisting of a sub-query.
-
-       Arguments:
-       - columns: a list of pairs; each first element is the alias for the
-         column created in the sub-query and the second element is the
-         expression to be computed.
-       - context: the context for the sub-query.
-       - alias: the name upon which the results of this sub-query can be
-         referred to in other parts of the query.
-    Sub_Query (columns : Vector (Pair Text Expression)) (context : Context) (alias : Text)
-
-# TODO Dubious constructor export
-from project.Internal.IR.Join_Kind import all
-from project.Internal.IR.Join_Kind export all
-
-## PRIVATE
-
-   The kind of the join operation.
-type Join_Kind
-    ## PRIVATE
-
-       Inner join.
-
-       The result will contain only rows that had a match in both left and right
-       source.
-    Join_Inner
-
-    ## PRIVATE
-
-       Left join.
-
-       The result will contain all rows from the left source. Some of them may
-       be duplicated if they have multiple matches on the right. If a row from
-       the left source has no match on the right, it will be present exactly
-       once in the result and the fields corresponding to the right source will
-       be set to NULL.
-    Join_Left
-
-    ## PRIVATE
-
-       Right join.
-
-       The result will contain all rows from the right source. Some of them may
-       be duplicated if they have multiple matches on the left. If a row from
-       the right source has no match on the left, it will be present exactly
-       once in the result and the fields corresponding to the left source will
-       be set to NULL.
-    Join_Right
-
-    ## PRIVATE
-
-       Cross join.
-
-       The result will contain a cross product of rows from the left source with
-       the right source. Its `on` list should be empty, instead `where_filters`
-       in the query can be used to filter the results.
-    Join_Cross
-
-# TODO Dubious constructor export
-from project.Internal.IR.Order_Descriptor import all
-from project.Internal.IR.Order_Descriptor export all
+import project.Internal.IR.Expression.Expression
 
 ## PRIVATE
 type Order_Descriptor
```

Comment:
Why `Order_Descriptor` and `Nulls_Order` did not get their own files like the rest of the types?

Reply:
Laziness - done now.